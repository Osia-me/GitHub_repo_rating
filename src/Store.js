import React, { useState, useEffect } from "react";
import axios from 'axios';

export const Context = React.createContext(null);

export const Store = ({children}) => {
  const [state, setState] = useState([]);
  const linkToLastWeek = "https://api.github.com/search/repositories?q=created:%3E2021-06-13&sort=stars&order=desc";
  const linkToReposByLang = (lang) => `https://api.github.com/legacy/repos/search/language=${lang}`

  useEffect(() => {
    axios
      .get(linkToLastWeek)
      .then((response) => {
        setState({
          items: response.data.items,
          fav: [],
          view: null,
          language: 'Select language'
        });
      })
      .catch((error) => {
        alert('Something went wrong!')
      });
  }, []);

  useEffect(() => {
    if(state.language && state.language !== 'Select language'){
      axios.get(linkToReposByLang(state.language)).then((response) => {
        setState({
          items: response.data.repositories.map( el => ({
            name: el.name,
            description:el.description,
            html_url: el.url,
            stargazers_count: Math.round(el.score),
            id: Math.random() * 2.5
          })),
          fav: [],
          view: null,
          language: state.language
        });
      })
      .catch((error) => {
        alert('Something went wrong!')
      });
    }
  },[state.language])

  return (
    <Context.Provider value={[state, setState]}>
      { children }
    </Context.Provider>
  );
};

export default Store;
