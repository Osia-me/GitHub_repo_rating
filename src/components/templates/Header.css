.header {
  width:auto;
  height:60px;
  margin-bottom: 14px;
  padding-left: 14px;
  padding-right: 14px;
  background-color: #2c333e;
  opacity:0.9;
  position:sticky;
  top:0;
  z-index: 1;
  -webkit-backdrop-filter: blur(8px);
  -moz-backdrop-filter: blur(8px);
  backdrop-filter: blur(8px);
  border-radius: 0px 0px 15px 15px;
}

.header-nav {
  width:100%;
  height:100%;
  display:flex;
  justify-content: space-between;
  align-items: center;
  color: white;
}

.header-name {
  margin: 1em;
  font-size: 1.5rem;
  font-weight: bold;
  vertical-align:8px;
}

.header-right {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
}

.header-language {
  border-color: white;
  padding: 5px;
  margin: 1em;
  cursor:pointer;
}
