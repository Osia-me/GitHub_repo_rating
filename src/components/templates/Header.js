import React, { useState, useEffect, useContext } from 'react';
import Star from '../elements/Star/Star';
import Logo from '../elements/Logo/Logo';
import Language from '../elements/Language/Language';
import { Context } from '../../Store.js';
import './Header.css'

function Header() {
  const [state, setState] = useContext(Context);
  const [clickStar, setClickStar] = useState(0);

  const StarClickHandler = () => {
    clickStar === 0 ? setClickStar(1) : setClickStar(0);
  }

  useEffect(() => {
    if(clickStar){
      setState({...state, view: 'starred'});
    } else{
      setState({...state, view: null});
    }
  },[clickStar, setState]);

  return (
      <header className='header'>
        <nav className='header-nav grow'>
          <Logo />
          <div className='header-right'>
            <Star width={35} height={35} starclick={StarClickHandler} color = {clickStar ? 'color' : null}/>
            <Language />
          </div>

        </nav>
      </header>
  );
}

export default Header;
