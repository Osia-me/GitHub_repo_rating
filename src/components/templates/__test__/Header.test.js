import Store from '../../../Store.js';
import { shallow } from 'enzyme';
import Header from "../Header";

test('renders Header without crashing', () => {
  const wrapper = shallow(
    <Store className="app">
      <Header/>
    </Store>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains(<Header />)).toBe(true);

});
