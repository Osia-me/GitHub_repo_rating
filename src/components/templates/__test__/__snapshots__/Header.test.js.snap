// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header without crashing 1`] = `ShallowWrapper {}`;
